### ISSUE TITLE
'npm run dev' is broken on Ubuntu 20.10

### ISSUE THREAD

**Issue by bisgardo** [Concordium/concordium-network-dashboard#52]
**Bug Description**

The development mode of the network dashboard doesn't get through startup on Ubuntu 20.10. It works on Ubuntu 20.04.

**Steps to Reproduce**

Run `npm run dev`.

**Expected Result**

A browser window with the compiled dashboard should open automatically.

**Actual Result**

The run script crashes with an error message that a tool named `gnome-open` cannot be found. The website cannot be manually opened afterwards either.

As `gnome-open` has been [defunct](https://askubuntu.com/questions/101965/what-is-the-replacement-for-gnome-open-in-gnome) for quite some time, and also isn't available for Ubuntu 20.04, the solution is not to get it installed.

Full output of `npm run dev`:

```
8:48:19 AM - Starting compilation in watch mode...
[server] 
[client] (node:1252921) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
[client] (Use `node --trace-deprecation ...` to show where the warning was created)
[client] ℹ ｢wds｣: Project is running at http://localhost:8090/
[client] ℹ ｢wds｣: webpack output is served from /public/
[client] ℹ ｢wds｣: Content not from webpack is served from /home/mo/work/network-dashboard
[server] 
[server] 8:48:21 AM - Found 0 errors. Watching for file changes.
[server] The App version is 1.0.0
[server] [HPM] Proxy created: /  -> http://localhost:8090/
[server] [HPM] Proxy created: /  -> http://localhost:8090/
[server] App listening on port 3001!
[server] WARNING: This is the development-only backend.
[server] See here for the real ones:
[server] https://gitlab.com/Concordium/p2p-client/blob/develop/src/bin/collector.rs
[server] https://gitlab.com/Concordium/p2p-client/blob/develop/src/bin/collector_backend.rs
[client] ℹ ｢wdm｣: Hash: 05232b4de718c14b87d4
[client] Version: webpack 4.44.1
[client] Time: 1846ms
[client] Built at: 08/10/2021 8:48:22 AM
[client]                      Asset      Size   Chunks                         Chunk Names
[client]                 index.html  1.35 KiB           [emitted]              
[client]    main-05232b4d-bundle.js  3.71 MiB     main  [emitted] [immutable]  main
[client] vendors-05232b4d-bundle.js  1.66 MiB  vendors  [emitted] [immutable]  vendors
[client] Entrypoint main = vendors-05232b4d-bundle.js main-05232b4d-bundle.js
[client] [0] multi (webpack)-dev-server/client?http://localhost:8090 (webpack)/hot/dev-server.js @babel/polyfill ./src/client/client 64 bytes {main} [built]
[client] [./node_modules/@babel/polyfill/lib/index.js] 686 bytes {vendors} [built]
[client] [./node_modules/@babel/polyfill/lib/noConflict.js] 567 bytes {vendors} [built]
[client] [./node_modules/core-js/library/fn/global.js] 87 bytes {vendors} [built]
[client] [./node_modules/webpack-dev-server/client/index.js?http://localhost:8090] (webpack)-dev-server/client?http://localhost:8090 4.29 KiB {vendors} [built]
[client] [./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.51 KiB {vendors} [built]
[client] [./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.53 KiB {vendors} [built]
[client] [./node_modules/webpack-dev-server/client/utils/createSocketUrl.js] (webpack)-dev-server/client/utils/createSocketUrl.js 2.91 KiB {vendors} [built]
[client] [./node_modules/webpack-dev-server/client/utils/log.js] (webpack)-dev-server/client/utils/log.js 964 bytes {vendors} [built]
[client] [./node_modules/webpack-dev-server/client/utils/reloadApp.js] (webpack)-dev-server/client/utils/reloadApp.js 1.59 KiB {vendors} [built]
[client] [./node_modules/webpack-dev-server/client/utils/sendMessage.js] (webpack)-dev-server/client/utils/sendMessage.js 402 bytes {vendors} [built]
[client] [./node_modules/webpack-dev-server/node_modules/strip-ansi/index.js] (webpack)-dev-server/node_modules/strip-ansi/index.js 161 bytes {vendors} [built]
[client] [./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {main} [built]
[client] [./node_modules/webpack/hot/dev-server.js] (webpack)/hot/dev-server.js 1.59 KiB {vendors} [built]
[client] [./src/client/client.ts] 657 bytes {main} [built]
[client]     + 332 hidden modules
[client] Child HtmlWebpackCompiler:
[client]      1 asset
[client]     Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
[client]     [./node_modules/html-webpack-plugin/lib/loader.js!./src/client/index.ejs] 1.63 KiB {HtmlWebpackPlugin_0} [built]
[client] ℹ ｢wdm｣: Compiled successfully.
[client] /home/mo/work/network-dashboard/node_modules/open-browser-webpack-plugin/index.js:46
[client]         if (err) throw err;
[client]                  ^
[client] 
[client] Error: Command failed: /home/mo/work/network-dashboard/node_modules/open/vendor/xdg-open "http://localhost:3001"
[client] /home/mo/work/network-dashboard/node_modules/open/vendor/xdg-open: 500: gnome-open: not found
[client] 
[client]     at ChildProcess.exithandler (child_process.js:308:12)
[client]     at ChildProcess.emit (events.js:314:20)
[client]     at maybeClose (internal/child_process.js:1051:16)
[client]     at Socket.<anonymous> (internal/child_process.js:442:11)
[client]     at Socket.emit (events.js:314:20)
[client]     at Pipe.<anonymous> (net.js:673:12) {
[client]   killed: false,
[client]   code: 4,
[client]   signal: null,
[client]   cmd: '/home/mo/work/network-dashboard/node_modules/open/vendor/xdg-open "http://localhost:3001"'
[client] }
[client] npm ERR! code ELIFECYCLE
[client] npm ERR! errno 1
[client] npm ERR! concordium-dashboard@1.0.1 dev-client: `webpack-dev-server -w`
npm ERR! Exit status 1
[client] npm ERR! 
[client] npm ERR! Failed at the concordium-dashboard@1.0.1 dev-client script.
[client] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[client] 
npm ERR! A complete log of this run can be found in:
[client] npm ERR!     /home/mo/.npm/_logs/2021-08-10T06_48_22_196Z-debug.log
[client] npm run dev-client exited with code 1

```

**Versions**
- Software Version: `main` branch.
- OS: Ubuntu 20.10 (running Gnome).
- Browser: Firefox

**Comment by bisgardo** [comment]
The root cause of the problem is that the (old and unmaintained) webpack plugin [`open-browser-webpack-plugin`](https://www.npmjs.com/package/open-browser-webpack-plugin) that we use [depends](https://github.com/baldore/open-browser-webpack-plugin/blob/master/package.json) on [`open`](https://www.npmjs.com/package/open) with ancient version 0.0.5 (fixed) which in turn ships an ancient version of [`xdg-open`](https://linux.die.net/man/1/xdg-open).

`xdg-open` is a shell script which includes the function
```
open_gnome()
{
    if gvfs-open --help 2>/dev/null 1>&2; then
        gvfs-open "$1"
    else
        gnome-open "$1"
    fi
   ...
}
```

So the problem is that `gvfs-open` -- which is now *also* defunct -- was removed in Ubuntu 20.10 and hence reanimated the fallback to `gnome-open`.

The proper solution is to either make `open-browser-webpack-plugin` depend on a newer version of `open`, override the dependency on our side (`yarn` [supports](https://yarnpkg.com/configuration/manifest/#resolutions) this), or migrate to webpack's [built in ability](https://webpack.js.org/configuration/dev-server/#devserveropen) to do this without plugins.

Until then, a working workaround is to note that in Ubuntu 20.04, `gvfs-open` was already deprecated and replaced by wrapper scripts for `gio`. I made a small repository [`gvfs-bin-gio`](https://github.com/bisgardo/gvfs-bin-gio) for building a small Debian package with the files. Building and installing that also fixes the problem.

**Comment by limemloh** [comment]
The `open-browser-webpack-plugin` is not really needed for anything other than saving the developer from typing in a simple address, and I think the best solution is just to remove it.

**Comment by abizjak** [comment]
Would you please do so @limemloh and close this issue?

**Comment by limemloh** [comment]
Fixed in #54
